### ISSUE TITLE
Летопись: новые записи автоматически писать в дневники всех героев

### ISSUE THREAD

**Issue by Tiendil** [Tiendil/the-tale#2071]
Можно даже без выделение обычной фразы в лингвистике, просто ввести особый тип заипис.

Писать для всех героев сразу при появлении записи.

**Comment by ArgoTheTale** [comment]
Лучше писать не сразу при появлении записи, а при заходе героя в город (или любой из городов, к которым относится запись). Это будет более правдоподобно. Герой пришёл в город, заглянул в трактир и собрал слухи, которые затем занёс в дневник, что и стало видно Хранителю. Запись вида:
"Разговорился с местным прядильщиком и выведал кое-что из свежих новостей. (текст летописи)"
Иначе получаются какие-то странные массовые коммуникации для пятнадцатого-то века.

**Comment by Tiendil** [comment]
Это правдоподобно, но сложнее реализовать на порядок. Придётся запоминать какой герой какие записи видел. Кроме того, для большинства это не будет заметно. 

А так "встретил гонцы - узнал новости".

**Comment by Vishoz** [comment]
Сложнее, да. Но если при посещении любого города, то не суперсложно.
Условно говоря, понадобится табличка на 3 столбца для связи героя, записи + статус, и триггер, который на входе в город будет ее чекать, постить записи и апдейтить статус.
Но будет на порядок реалистичнее, особенно если много законов было принято подряд.
Герой встретил гонца, гонца, гонца... ))))

**Comment by Tiendil** [comment]
Лезть в таблицу при каждом посещении города героем для нас непозволительно дорого.
Размер таблицы будет равен <количество героев>*<размер летописи>. Это само по себе плохо.

**Comment by Vishoz** [comment]
Согласен. Смотрю с архитектуры систем в которых десятки миллиардов записей )
Есть вариант, запоминать последнее время посещения героем города и по триггеру выкатывать записи, которые были приняты до посещения нового города, это дешевле.

**Comment by Tiendil** [comment]
Делать в рамках gh-2116

Пусть сервис подписывается на amqp сообщение "герой пришёл в город"  и по нему добавляет запись в дневник.

**Comment by SilentWrangler** [comment]
Альтернатива: герой узнаёт новости исключительно **родного** города и Соратника/Противника. Навесить указатель на "последнюю услышанную новость". При взятии квеста высвечивается самая ранняя из неуслышанных вместо "открыли новый магазинчик", при выполнении высвечивается весь стак.

**Comment by Tiendil** [comment]
Сделать как заготовку функциональности для «молитв героев»?
